File: erpnext/quality_management/report/resumen_programa_anual_sst_actividades/resumen_programa_anual_sst_actividades.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Resumen Programa Anual SST Actividades"] = {
	"filters": [
		{
			"fieldname":"politics",
			"label": __("Politicas"),
			"fieldtype": "Link",
			"options": "Politicas SSOMA",
			"default": "",
			"width": "100px"
		},
		{
			"fieldname":"mes",
			"label": __("Mes"),
			"fieldtype": "Select",
			"default": "",
			"options": ["","enero", "febrero", "marzo", "abril", "mayo", "junio","julio","agosto","setiembre","octubre","noviembre","diciembre"],
			"width": "100px"
		},
		{
			"fieldname":"fecha",
			"label": __("Fecha"),
			"fieldtype": "Date",
			"options": "Politicas SSOMA",
			"default": "",
			"width": "100px"
		},
		{
			"fieldname":"estado",
			"label": __("Estado"),
			"fieldtype": "Select",
			"default": "",
			"options": ["","Programado", "Ejecutado", "Reprogramado", "En Proceso"],
			"width": "100px"
		},
	],

};

File: erpnext/quality_management/report/resumen_programa_anual_sst_actividades/resumen_programa_anual_sst_actividades.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

# import frappe
import requests

def execute(filters=None):
	columns, data = [], []
	columns = get_columns()
	arrayData = requests.post("https://recursoshumanos.shalom.com.pe/api/reporte/reportProgramaActualSST", json = filters)
	arrayData = arrayData.json()
	if "data" in arrayData:
		return columns, arrayData.get("data")


def get_columns():
	columns = [
		{
			'label': "Lista de actividades",
			'fieldname': 'actividad',
			'fieldtype': 'Data',
			'width': 300
		},
		{
            'label': "Indicador",
            'fieldname': 'indicador',
            'fieldtype': 'Data',
            'width': 300
        },
		{
			'label': "Mes de Ejecución",
			'fieldname': 'mes_ejecucion',
			'fieldtype': 'Data',
			'width': 100,
		},
		{
			'label': "Total Programadas",
			'fieldname': 'total_programados',
			'fieldtype': 'Data',
			'width': 100,
		},
		{
			'label': "Total Ejecutadas",
			'fieldname': 'total_ejecutados',
			'fieldtype': 'Data',
			'width': 100,
		},
		{
			'label': "Total Reprogramadas",
			'fieldname': 'total_reprogramados',
			'fieldtype': 'Data',
			'width': 100,
		},
		{
			'label': "% DE EFICIENCIA",
			'fieldname': 'eficiencia',
			'fieldtype': 'Data',
			'width': 100,
		},

	]
	return columns